core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.ts

```diff
@@ -170,6 +182,171 @@ export class ContributionAndReviewService {
+      explorationId: string
+  ): Promise<FetchSuggestionsResponse> {
+    if (shouldResetOffset) {
+      fetcher.offset = 0;
+      fetcher.suggestionIdToDetails = {};
+    }
+
+    const explorationBackendResponse = await this.
+      readOnlyExplorationBackendApiService.fetchExplorationAsync(
+        explorationId, null);
+
+    const responseBody = await this.
+      contributionAndReviewBackendApiService.
+      fetchSuggestionsAsync(
+        fetcher.type,
+        null,
+        fetcher.offset,
+        fetcher.sortKey,
+        explorationId
+      );
+
+    const explorationBackendDict: ExplorationBackendDict = {
+      auto_tts_enabled: explorationBackendResponse.auto_tts_enabled,
+      correctness_feedback_enabled: explorationBackendResponse.
+        correctness_feedback_enabled,
+      draft_changes: [],
+      init_state_name: explorationBackendResponse.exploration.init_state_name,
+      states: explorationBackendResponse.exploration.states,
+      param_changes: explorationBackendResponse.exploration.param_changes,
+      param_specs: explorationBackendResponse.exploration.param_specs,
+      title: explorationBackendResponse.exploration.title,
+      language_code: explorationBackendResponse.exploration.language_code,
+      next_content_id_index: explorationBackendResponse.
+        exploration.next_content_id_index,
+      exploration_metadata: explorationBackendResponse.exploration_metadata,
+      is_version_of_draft_valid: false,
+      draft_change_list_id: explorationBackendResponse.draft_change_list_id
+    };
+
+    const exploration: Exploration = this.explorationObjectFactory.
+      createFromBackendDict(explorationBackendDict);
+    const states = this.getStatesForExploration(exploration);
+    const sortedTranslationCards = this.sortTranslationSuggestionsByState(
+      responseBody.suggestions,
+      // exploration.getStates(),
+      states,
+      explorationBackendDict.init_state_name
+    );
+
+    responseBody.suggestions = sortedTranslationCards;
+    const responseSuggestionIdToDetails = fetcher.suggestionIdToDetails;
+    fetcher.suggestionIdToDetails = {};
+    const targetIdToDetails = responseBody.target_id_to_opportunity_dict;
+    responseBody.suggestions.forEach((suggestion) => {
+      const suggestionDetails = {
+        suggestion: suggestion,
+        details: targetIdToDetails[suggestion.target_id]
+      };
+
+      responseSuggestionIdToDetails[suggestion.suggestion_id] = (
+        suggestionDetails);
+    });
+    return {
+      suggestionIdToDetails: responseSuggestionIdToDetails,
+      more: Object.keys(fetcher.suggestionIdToDetails).length > 0
+    };
+  }
+
+  private getStatesForExploration(exploration: Exploration) {
+    return exploration.getStates();
+  }
+
+  // Function to sort translation cards by state.
+  sortTranslationSuggestionsByState(
+      // eslint-disable-next-line
+      translationSuggestions: any[],
+      states: States,
+      initStateName: string | null
```

Comment:
How can this be `null`? Don't all explorations have an `initStateName`?

Reply:
In exploration object initStateName can be a null or a string but for `computeGraphService` method it has to be of string type. Thus to avoid assertion error, I added a if condition at line 238.